### PR TITLE
removing flake8 in favor of ruff

### PR DESCRIPTION
**Pull request by danielbeach**
Remove flake8 and replace it with ruff.
- change the git workflow to use ruff
- add ruff to the pre-commit hooks
- add ruff to the pyproject.toml

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = "^3.9"
+deltalake = "^0.7.0"
```
Comment (MrPowers) [inline pyproject.toml:10]: Don't think we want to add deltalake as a dependency for this particular project.

Comment (danielbeach) [reply pyproject.toml]: @MrPowers Yeah sorry, I will remove it. I was just putting this in WIP so I could test the `ruff` git workflow.

Comment (MrPowers) [reply pyproject.toml]: @danielbeach - yea, keep iterating.  I normally have to tweak these workflows like 10 times before I get them working, haha.  Let me know if you need any help.

**Comment by robertkossendey** [comment]
Could we also change the badge in the README? Otherwise great PR, thank you very much!

**Comment by danielbeach** [comment]
@MrPowers @robertkossendey I checked the flake8 config https://github.com/MrPowers/mack/blob/main/.github/workflows/flake8.yml and it appears the `line-length` was about the only "custom" config I could find, which I migrated to flake8. Let me know if there's anything else we should add.

**Comment by robertkossendey** [comment]
@danielbeach great, thank you! I think we should have a separate discussion of what kind of rules we want to introduce, so this should be fine!

cc: @MrPowers

File: pyproject.toml [hunk 2]
```diff
@@ -7,6 +7,7 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = "^3.9"
+ruff = "^0.0.254"
```
Comment (MrPowers) [inline pyproject.toml:10]: Think this needs to be a dev-dependency.

**Comment by MrPowers** [comment]
@danielbeach - PR looks great.  Think we just need to make ruff a dev-dependency.  That way users don't get ruff installed as a transitive dependency when they run `pip install mack`.  Really excited about this!

File: pyproject.toml (hunk 2 above)
Comment (danielbeach) [reply pyproject.toml]: @MrPowers fixed.

**Review by robertkossendey — APPROVED** [review]

**Comment by MrPowers** [comment]
Thanks for adding this.  Merging now.  Fingers crossed it all works!

**Comment by MrPowers** [comment]
@danielbeach - BTW, pro tip (aka pro hack) for Poetry.  Whenever you're dealing with annoying `poetry.lock` issues because you changed the `pypoetry.toml` file, just run `rm poetry.lock` and then regenerate it with `poetry install`.  It's so annoying haha, but that's the best way I've found.

**Pull request merged**